File: 007-Ruby/servidor.rb
require 'webrick'
require 'json'

json_data = { mensaje: 'Hola mundo desde el servidor' }

server = WEBrick::HTTPServer.new(Port: 8000)

server.mount_proc '/' do |req, res|
  # Set CORS headers
  res['Access-Control-Allow-Origin'] = '*' # Allow requests from any origin
  res['Access-Control-Allow-Methods'] = 'GET, POST, PUT, DELETE, OPTIONS' # Allow specific HTTP methods
  res['Access-Control-Allow-Headers'] = 'Content-Type' # Allow specific headers
  res['Access-Control-Max-Age'] = '3600' # Cache preflight response for 1 hour
  res['Content-Type'] = 'application/json'
  res.body = json_data.to_json
end

trap('INT') { server.shutdown }

server.start

File: 002-Nodejs/servidor.js
const http = require('http');

const server = http.createServer((req, res) => {
  res.setHeader('Content-Type', 'application/json');
  res.setHeader('Access-Control-Allow-Origin', '*');
  const responseData = JSON.stringify({ mensaje: 'Hola mundo desde el servidor' });
  res.end(responseData);
});

const PORT = 3000; 
server.listen(PORT, () => {
  console.log(`Server running at http://localhost:${PORT}/`);
});